libraries/protocol/worker_proposal_operations.cpp

```diff
@@ -0,0 +1,29 @@
+#include <golos/protocol/worker_proposal_operations.hpp>
+#include <golos/protocol/exceptions.hpp>
+#include <golos/protocol/validate_helper.hpp>
+
+namespace golos { namespace protocol {
+
+    static inline void validate_account_name(const string& name) {
+        GOLOS_CHECK_VALUE(is_valid_account_name(name), "Account name ${name} is invalid", ("name", name));
+    }
+
+    /// TODO: after the hardfork, we can rename this method validate_permlink because it is strictily less restrictive than before
+    ///  Issue #56 contains the justificiation for allowing any UTF-8 string to serve as a permlink, content will be grouped by tags
+    ///  going forward.
+    inline void validate_permlink(const string& permlink) {
+        GOLOS_CHECK_VALUE(permlink.size() < STEEMIT_MAX_PERMLINK_LENGTH, "permlink is too long");
+        GOLOS_CHECK_VALUE(fc::is_utf8(permlink), "permlink not formatted in UTF8");
+    }
```

Comment:
Move `validate_account_name` and `validate_permlink` to common module, instead of copy-paste.

Reply:
https://github.com/GolosChain/golos/issues/1033